spring/boot-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/NettyBasedArmeriaBufferFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.spring.web.reactive;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.ByteBuffer;
+
+import org.springframework.core.io.buffer.DataBuffer;
+import org.springframework.core.io.buffer.DataBufferFactory;
+import org.springframework.core.io.buffer.NettyDataBuffer;
+import org.springframework.core.io.buffer.NettyDataBufferFactory;
+
+import com.linecorp.armeria.common.DefaultHttpData;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.unsafe.ByteBufHttpData;
+
+import io.netty.buffer.ByteBuf;
+
+/**
+ * A factory which forwards the method calls defined in {@link DataBufferFactory} to the given
+ * {@link NettyDataBufferFactory} instance.
+ */
+class NettyBasedArmeriaBufferFactory extends AbstractArmeriaBufferFactory<NettyDataBufferFactory> {
+
+    NettyBasedArmeriaBufferFactory(NettyDataBufferFactory delegate) {
+        super(delegate);
+    }
+
+    @Override
+    protected DataBuffer wrap0(ByteBufHttpData httpData) {
+        return delegate().wrap(httpData.content());
```

Comment:
How and when the `ByteBuf` returned by `httpData.content()` is released?

Reply:
Spring framework will consume it if a user wants to read HTTP body, then release the `DataBuffer` which is returned here using https://github.com/spring-projects/spring-framework/blob/master/spring-core/src/main/java/org/springframework/core/io/buffer/DataBufferUtils.java#L406.

According to the code, the `DataBuffer` instance, which holds a `ByteBuf`, should be a subclass of `PooledDataBuffer`. 

So, I left comment on https://github.com/line/armeria/pull/1326/files#diff-4938ea1a82e4c011c7439b3a255b8a16R89 and made the return type `PooledDataBuffer` explicitly.